### DISCUSSION TITLE
How to unit test tracing setup?

### DISCUSSION THREAD

**Discussion by jaads** [tokio-rs/tracing#3100] [Q&A]
I'm a big fan of automated tests instead of manual testing something. 

since `tracing` has a lot to offer and configure, I wonder how users of the lib can unit test their configuration. are there some examples?

**Comment by hds** [comment]
What part of the configuration do you want to test? A lot can be tested by `tracing-mock`, this is an (as yet) unpublished crate that you can find in this repo.

But depending on what bits you would like to test, it may not suit your use case.
